Download transfers for multiple addresses

dshulyak opened the pull request:
one thing that i want to add but lack some info is how to support adding accounts in runtime. i will add that once multi-accounts will be implemented in status-go

dshulyak committed:
Refactor logic from reactor

dshulyak committed:
Rework downloaders so that can accept multiple accounts

ghost commented:
<!--prchecklist--> 
### Pull Request Checklist
- [ ] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://docs.status.im/docs/build_status_go.html))?

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (12)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | b5ce9030 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1480/1/) | 2019-06-04 08:58:15 | ~1 min | `linux` | [:package: zip](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-190604-085713-b5ce90.zip) |
| :x: | b5ce9030 | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1480/1/) | 2019-06-04 08:58:19 | ~1 min | `ios` | [:page_facing_up: log](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1480/1/consoleText) |
| :x: | b5ce9030 | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1480/1/) | 2019-06-04 08:59:19 | ~2 min | `android` | [:page_facing_up: log](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1480/1/consoleText) |
| | | | | | | |
| :heavy_check_mark: | d290be4c | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1480/2/) | 2019-06-05 08:24:57 | ~47 sec | `linux` | [:package: zip](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-190605-082409-d290be.zip) |
| :heavy_check_mark: | d290be4c | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1480/2/) | 2019-06-05 08:28:06 | ~3 min | `ios` | [:package: zip](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-190605-082408-d290be.zip) |
| :heavy_check_mark: | d290be4c | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1480/2/) | 2019-06-05 08:30:52 | ~6 min | `android` | [:package: aar](https://status-go.ams3.digitaloceanspaces.com/status-go-android-190605-082406-d290be.aar) |
| | | | | | | |
| :heavy_check_mark: | 8321d3be | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1480/3/) | 2019-06-05 10:03:43 | ~46 sec | `linux` | [:package: zip](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-190605-100257-8321d3.zip) |
| :heavy_check_mark: | 8321d3be | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1480/3/) | 2019-06-05 10:07:09 | ~4 min | `ios` | [:package: zip](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-190605-100255-8321d3.zip) |
| :heavy_check_mark: | 8321d3be | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1480/3/) | 2019-06-05 10:09:07 | ~6 min | `android` | [:package: aar](https://status-go.ams3.digitaloceanspaces.com/status-go-android-190605-100254-8321d3.aar) |
| | | | | | | |
| :heavy_check_mark: | 4f11fc9c | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1480/4/) | 2019-06-06 05:49:31 | ~39 sec | `linux` | [:package: zip](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-190606-054852-4f11fc.zip) |
| :heavy_check_mark: | 4f11fc9c | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1480/4/) | 2019-06-06 05:51:46 | ~2 min | `ios` | [:package: zip](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-190606-054851-4f11fc.zip) |
| :heavy_check_mark: | 4f11fc9c | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1480/4/) | 2019-06-06 05:55:27 | ~6 min | `android` | [:package: aar](https://status-go.ams3.digitaloceanspaces.com/status-go-android-190606-054850-4f11fc.aar) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 21234f48 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1480/5/) | 2019-06-06 05:50:41 | ~32 sec | `linux` | [:package: zip](https://status-go.ams3.digitaloceanspaces.com/status-go-desktop-190606-055008-21234f.zip) |
| :heavy_check_mark: | 21234f48 | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1480/5/) | 2019-06-06 05:54:38 | ~2 min | `ios` | [:package: zip](https://status-go.ams3.digitaloceanspaces.com/status-go-ios-190606-055147-21234f.zip) |
| :heavy_check_mark: | 21234f48 | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1480/5/) | 2019-06-06 06:01:27 | ~5 min | `android` | [:package: aar](https://status-go.ams3.digitaloceanspaces.com/status-go-android-190606-055527-21234f.aar) |
| | | | | | | |
| :x: | 9fc85390 | [#6](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1480/6/) | 2019-06-13 06:26:35 | ~10 sec | `android` | [:page_facing_up: log](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-1480/6/consoleText) |
| :x: | 9fc85390 | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1480/6/) | 2019-06-13 06:26:37 | ~9.2 sec | `linux` | [:page_facing_up: log](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-1480/6/consoleText) |
| :x: | 9fc85390 | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1480/6/) | 2019-06-13 06:26:46 | ~18 sec | `ios` | [:page_facing_up: log](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-1480/6/consoleText) |

dshulyak committed:
Support for downloading transfers for multiple accounts

dshulyak committed:
Update readme with minor fixes

dshulyak commented:
@yenda you can use this change. i added all api changes to readme. while multi accounts are not implemented wallet service will be using single address for all transfers, but api should stay the same.

dshulyak committed:
Refactor issues reported by linter

dshulyak committed:
Rewrite concurrent downloader and make erc20 downloader concurrent

dshulyak committed:
Generate unique ID for every transfer

dshulyak committed:
Write better tests for reorg handling and improve handling description

dshulyak committed:
Renaming leftover

dshulyak committed:
Add logs and simplify lastKnownHeader

dshulyak committed:
Update accounts with all blocks in the set

dshulyak committed:
Encrypt database with same password taht is used for pfs db

dshulyak committed:
Rename ConcurrentRunner to AtomicGroup

dshulyak committed:
Add fast catchup after app was offline for longer than 15 blocks

When downloader is started it will go through following procedure:
- verify that the last synced block is still in canonical chain
- run fast indexer for every account from last synced block to head of the chain
- start watching block by block

dshulyak committed:
Remove ununes isSequence

dshulyak committed:
Increase timeout in test: bootstrap procedure may take longer now

dshulyak committed:
Remove unused code in database and adjust documentation
